js/private/npm_utils.bzl

```diff
@@ -3,11 +3,11 @@
 def _bazel_name(name, version):
     "Make a bazel friendly name from a package name and a version that can be used in repository and target names"
     escaped = name.replace("/", "_").replace("@", "at_")
-    return "%s_%s" % (escaped, version)
+    return "%s_%s" % (escaped, version.replace("@", "at_").replace("+", "_"))
 
 def _versioned_name(name, version):
     "Make a developer-friendly name for a package name and version"
-    return "%s@%s" % (name, version)
+    return "%s@%s" % (name, version.replace("@", "at_").replace("+", "_"))
```

Comment:
Using peer deps (the pnpm logger package) revealed a new case where invalid repository names were being created. This fixes it, but you might want to double check that what I'm doing here is in line with your intention for these methods.